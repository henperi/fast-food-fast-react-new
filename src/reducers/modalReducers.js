const modalDefaultState = {
  id: undefined,
  isOpen: false,
};

const modalReducer = (state = modalDefaultState, action) => {
  switch (action.type) {
    case 'OPEN_MODAL':
      return {
        ...action.modal,
      };
    case 'CLOSE_MODAL':
      return {
        ...state,
        isOpen: action.modal,
      };

    default:
      return state;
  }
};

export default modalReducer;
