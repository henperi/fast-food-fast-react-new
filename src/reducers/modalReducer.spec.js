/* eslint-disable no-undef */
import modalReducers from './modalReducers';

describe('modalReducers', () => {
  const initialState = {
    id: undefined,
    isOpen: false,
  };
  it('should return the default state if no action is specified', () => {
    const action = { type: '' };
    expect(modalReducers(initialState, action)).toMatchSnapshot({});
    expect(modalReducers(initialState, action)).toEqual(initialState);
  });

  it('should update the state and SET_MENU when type is SET_MENU and payload is specified', () => {
    const action = {
      type: 'OPEN_MODAL',
      modal: [{ user_id: 5, name: 500 }],
    };
    expect(modalReducers(initialState, action)).toMatchSnapshot({});
    expect(modalReducers(initialState, action)).toEqual({ 0: { name: 500, user_id: 5 } });
  });

  it('should update the state and SET_MENU when type is SET_MENU and payload is specified', () => {
    const action = {
      type: 'CLOSE_MODAL',
      modal: [{ user_id: 5, name: 500 }],
    };
    expect(modalReducers(initialState, action)).toMatchSnapshot({});
    expect(modalReducers(initialState, action)).toEqual({
      id: undefined,
      isOpen: [{ name: 500, user_id: 5 }],
    });
  });
});
