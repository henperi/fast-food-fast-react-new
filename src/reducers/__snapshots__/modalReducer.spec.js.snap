// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modalReducers should return the default state if no action is specified 1`] = `
Object {
  "id": undefined,
  "isOpen": false,
}
`;

exports[`modalReducers should update the state and SET_MENU when type is SET_MENU and payload is specified 1`] = `
Object {
  "0": Object {
    "name": 500,
    "user_id": 5,
  },
}
`;

exports[`modalReducers should update the state and SET_MENU when type is SET_MENU and payload is specified 2`] = `
Object {
  "id": undefined,
  "isOpen": Array [
    Object {
      "name": 500,
      "user_id": 5,
    },
  ],
}
`;
